src/NuGet.Core/NuGet.Packaging/Signing/Signatures/Signature.cs

```diff
@@ -122,6 +133,27 @@ private static SignatureType GetSignatureType(SignerInfo signer)
             // TODO: Change this to use the new attributes that justin is adding.
             return SignatureType.Author;
         }
+        private static IReadOnlyList<Timestamp> GetTimestamps(SignerInfo signer)
+        {
+            var authorUnsignedAttributes = signer.UnsignedAttributes;
+            
+            var timestampList = new List<Timestamp>();
+
+            foreach (var attribute in authorUnsignedAttributes)
+            {
+                if (string.Equals(attribute.Oid.Value, Oids.SignatureTimeStampTokenAttributeOid, StringComparison.Ordinal))
+                {
+                    foreach (var value in attribute.Values)
```

Comment:
@mishra14 correct me if I'm wrong, but I think for a timestamp we will always have only one attribute, therefore is unnecessary to do this for loop, we could just take the first element.

Reply:
@mishra14 @PatoBeltran I'm okay either way, earlier feedback suggested that there could/would be more than one.

Reply:
There could be multiple attributes with the timestamp Oid, but I'm not sure if there could be multiple values in an attribute

Reply:
Happy to make the appropriate fix once someone confirms :)

Reply:
looking at this other [code ](https://github.com/NuGet/NuGet.Client/blob/fd2e646127e074fcf007a961e59e33106ebdb4d0/src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs#L48) where we read timestamp, it seams, returning the first is fine.

Sorry about the confusion. You can just add the first entry to `Signature.Timestamps` and return to keep it consistent and if we add support to allow multiple timestamps, this will need minimal changes.

Reply:
Edit - 

I think adding multiple timestamps are fine. But just like [here](https://github.com/NuGet/NuGet.Client/blob/fd2e646127e074fcf007a961e59e33106ebdb4d0/src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs#L48), read attribute.Values[0].

Reply:
Ok, fixed.

One thing here: https://github.com/NuGet/NuGet.Client/blob/fd2e646127e074fcf007a961e59e33106ebdb4d0/src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs#L67

If there are multiple timestamps, seems like the last one wins? Should that loop explicitly track the most or least trusted one based on the intended logic?